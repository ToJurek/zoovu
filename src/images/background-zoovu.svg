<?xml version="1.0" encoding="utf-8"?>
<svg id="master-artboard" viewBox="0 0 1920 1080" xmlns="http://www.w3.org/2000/svg" width="1920px" height="1080px" version="1.1" style="shape-rendering:geometricPrecision; text-rendering:geometricPrecision; image-rendering:optimizeQuality; fill-rule:evenodd; clip-rule:evenodd"><defs><linearGradient id="gradient-0" gradientUnits="userSpaceOnUse" x1="922.131591796875" y1="11.360533714294434" x2="922.131591796875" y2="1091.3604736328125" gradientTransform="matrix(0.9998956037436877, 0.014480596334008552, -0.014025406994830817, 0.9684431656975776, 15.403020085386459, 21.087154293514615)">
                <stop offset="0" style="stop-color: #ffffff"/>
                <stop offset="1" style="stop-color: rgb(248, 248, 248);"/>
              </linearGradient><linearGradient id="gradient-1" gradientUnits="userSpaceOnUse" x1="960" y1="0" x2="960" y2="1080">
                <stop offset="0" style="stop-color: rgb(255, 255, 255);"/>
                <stop offset="1" style="stop-color: rgb(231, 231, 231);"/>
              </linearGradient></defs><rect id="ee-background" x="0" y="0" width="1920" height="1080" style="fill: url('#gradient-1'); fill-opacity: 1; pointer-events: none;"/>
 
 <metadata id="CorelCorpID_0Corel-Layer"/>
<g transform="matrix(4.453287124633789, 0, 0, 4.453287124633789, -2343.0559479637204, 568.8000087229798)"><g id="g-1" transform="matrix(1, 0, 0, 1, 0, 0)">
	<g id="g-2">
		<path class="st0" d="M652,46c27.6,0,50,22.4,50,50s-22.4,50-50,50s-50-22.4-50-50S624.4,46,652,46z" style="fill: rgb(255, 255, 255);"/>
	</g>
</g></g><g transform="matrix(-3.7170605659484863, 3.2741501331329346, -3.2741501331329346, -3.7170605659484863, 2903.47844935734, 1449.7246358820635)"><g id="g-5" transform="matrix(1, 0, 0, 1, 0, 0)">
	<g id="g-6">
		<path class="st0" d="M103,336.5L53,424h100L103,336.5" style="fill: rgb(255, 255, 255);"/>
	</g>
</g></g><g transform="matrix(4.453287124633789, 0, 0, 4.453287124633789, -1970.5298249168459, 244.04774675623378)"><g id="g-3" transform="matrix(1, 0, 0, 1, 0, 0)">
	<g id="g-4">
		<path class="st0" d="M652,46c27.6,0,50,22.4,50,50s-22.4,50-50,50s-50-22.4-50-50S624.4,46,652,46z" style="fill: rgb(255, 255, 255);"/>
	</g>
</g></g><foreignObject class="multi-line-input__container" x="823.1947631835938" y="441" height="33" width="224" transform="matrix(1, 0, 0, 1, 666.485948628404, -374.898346103477)"><div xmlns="http://www.w3.org/1999/xhtml" class="multi-line-input" style="font-size: 24px; font-family: 'Open Sans', sans-serif; fill: rgb(51, 51, 51); white-space: pre; color: rgb(51, 51, 51); display: inline-block;"></div></foreignObject><g transform="matrix(2.1829447746276855, -1.9545382261276245, 1.9545382261276245, 2.1829447746276855, 1125.011962890625, 53.6534522132813)"><g id="g-7" transform="matrix(1, 0, 0, 1, 0, 0)">
	<g id="g-8">
		<path class="st0" d="M58,49h100v100H58V49z" style="fill: rgb(255, 255, 255);"/>
	</g>
</g></g><g transform="matrix(2.4916787147521973, -1.9422504901885989, 1.9422504901885989, 2.4916787147521973, -182.17283429845384, 1156.4943715120926)"><g id="g-11" transform="matrix(1, 0, 0, 1, 0, 0)">
	<g id="g-12" transform="matrix(1, 5.551115123125783e-17, -5.551115123125783e-17, 1, 0.2084605848737624, 1.6822933988969453)">
		<path class="st0" d="M652,46c27.6,0,50,22.4,50,50s-22.4,50-50,50s-50-22.4-50-50S624.4,46,652,46z" style="fill: rgb(255, 255, 255);"/>
	</g>
</g></g></svg>