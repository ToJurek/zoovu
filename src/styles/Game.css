.row:after {
    content: "";
    display: table;
    clear: both;
}

.row {
    display: flex;
    width: 100%;
    font-size: 20px;

}

.game {
    width: 70vm;
}

.instruction {
    text-align: left;
    margin: 30px 20px 15px;
    font-size: 16px !important;
    color: #D1D1D1;
}