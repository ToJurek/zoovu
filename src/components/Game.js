import React from "react";
import "../styles/Game.css"
import Header from "./Header";
import {o_zoovu, u_zoovu, v_zoovu, z_zoovu} from "../images/scripts/images";
import Droppable from "./Droppable";
import styled from "styled-components"
import Draggable from "./Draggable";
import "../styles/InputFields.css"
import Message from "./Message";

const Item = styled.div`
  padding:8px;
  color: #555;
  transparancy: 1;
  border-radius: 3px
`;

const droppableStyle = {
    float: 'left',
    width: '150px',
    height: '150px',
    position: 'relative',
    flex: '50%',
    margin: '20px',
    borderRadius: '28px',
    border: 'green dashed 2px',
    background: '#EAEAEA'
}

const draggableStyle = {
    float: 'left',
    width: '150px',
    height: '150px',
    position: 'relative',
    flex: '50%',
    margin: '20px',
    borderRadius: '28px',
    border: 'solid #7A7A7A 2px',
    background: 'white'
}

const imagesPosition = [
    {imgID: 1, img: z_zoovu, dragFieldID: 1, dropFieldID: null},
    {imgID: 2, img: o_zoovu, dragFieldID: 2, dropFieldID: null},
    {imgID: 3, img: o_zoovu, dragFieldID: 3, dropFieldID: null},
    {imgID: 4, img: v_zoovu, dragFieldID: 4, dropFieldID: null},
    {imgID: 5, img: u_zoovu, dragFieldID: 5, dropFieldID: null}
]

const shuffle = (sourceArray) => {
    for (var i = 0; i < sourceArray.length - 1; i++) {
        var j = i + Math.floor(Math.random() * (sourceArray.length - i));

        var temp = sourceArray[j];
        sourceArray[j] = sourceArray[i];
        sourceArray[i] = temp;
    }
    var index = 0
    var updateArray = sourceArray.map(img => {
        index++
        return {imgID: img.imgID, img: img.img, dragFieldID: index, dropFieldID: img.dropFieldID}
    })
    return updateArray;
}

const mixedArray = shuffle(imagesPosition)

class Game extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            time: 0,
            isOn: false,
            isCorrect: false,
            imagesPosition: mixedArray

        }
        this.startTimer = this.startTimer.bind(this)
        this.wrongMatch = this.wrongMatch.bind(this)
        this.stopTimer = this.stopTimer.bind(this)
    }

    wrongMatch() {
        this.setState({
            time: this.state.time + 10
        })
    }

    timer = () => {
        this.timer = setInterval(() => this.setState({
            time: this.state.time + 1
        }), 1000);
    }

    startTimer = () => {
        if (!this.state.isOn) {
            this.setState({
                isOn: true,
                time: 0
            })
            this.timer()
        }
    }

    matchFields = (itemID, dropId) => {
        if (itemID === dropId || (itemID === 2 && dropId === 3) || (itemID === 3 && dropId === 2)) {
            const matches = this.state.imagesPosition
            var index = matches.findIndex(id => {
                return id.imgID === itemID
            })
            matches[index].dropFieldID = dropId
            this.setState({
                imagesPosition: matches
            })
            this.checkResults()
        } else {
            this.wrongMatch()
        }

    }

    checkResults() {
        let isCorrect = this.state.imagesPosition.filter(img => {
            return !img.dropFieldID
        })
        if (isCorrect.length === 0) {
            this.stopTimer()
        }
    }

    stopTimer() {
        this.setState({
            isOn: false,
            isCorrect: true
        })
        clearInterval(this.timer)
    }

    renderGame() {
        return (
            <div className={"game"}>
                <Header name={this.props.name} time={this.state.time}/>
                <div className={"row"}>
                    {this.state.imagesPosition.map(item => {
                        return (<Droppable returnItem={(item) => console.log(item)} style={draggableStyle}
                                           id={"drag" + item.imgID}><Draggable
                            id={"item" + item.imgID}><Item><img onDragStart={this.startTimer} className={"img-c"}
                                                                id={"item" + item.imgID}
                                                                src={item.img}/></Item></Draggable></Droppable>)
                    })}
                </div>
                <div className={"row"}>
                    <div className={"instruction"}>
                        <span>...and drop them here to make the logo great again!</span>
                    </div>
                </div>
                <div className={"row"}>
                    <div className={"inputs"}>
                        {[1, 2, 3, 4, 5].map(dropID => {
                            return (
                                <Droppable returnItem={(item) => this.matchFields(item, dropID)} id={"drop" + dropID}
                                           style={droppableStyle}></Droppable>)
                        })}
                    </div>
                </div>
            </div>
        )
    }


    render() {
        return (
            <div>
                {this.state.isCorrect ? <Message name={this.props.name} time={this.state.time}/> : this.renderGame()}
            </div>)


    }


}

export default Game