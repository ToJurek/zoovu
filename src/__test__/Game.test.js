import React from "react"
import Enzyme, {shallow, mount} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from "react-test-renderer";
import Game from "../components/Game";
import Droppable from "../components/Droppable";
import Draggable from "../components/Draggable";
import {o_zoovu, u_zoovu, v_zoovu, z_zoovu} from "../images/scripts/images";

import shuffle from "../components/Game"

Enzyme.configure({adapter: new Adapter()})

var sinon = require('sinon');

describe("Game Componnent", () => {
    it('render correctly', () => {
        const tree = renderer
            .create(<Game renderGamePage={true} name={"name"}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('check number of components before win', function () {
        const wrapper = shallow(<Game renderGamePage={true} name={"name"}/>)
        const images = wrapper.find("img").length
        const droppableArea = wrapper.find(Droppable).length
        const draggableArea = wrapper.find(Draggable).length
        const rows = wrapper.find(".row").length
        expect(images).toEqual(5)
        expect(droppableArea).toEqual(10)
        expect(draggableArea).toEqual(5)
        expect(rows).toEqual(3)
    });

    it('check number of components after win', function () {
        const wrapper = mount(<Game renderGamePage={false} name={"name"}/>)
        const rows = wrapper.find(".row")
        expect(rows.length).toEqual(4)
        expect(wrapper.props().name).toEqual('name')
        expect(wrapper.props().renderGamePage).toEqual(false)
    });

    it('should shuffle images', function () {
        const wrapper = shallow(<Game renderGamePage={true} name={"name"}/>)
        const imagesPosition = [
            {imgID: 1, img: z_zoovu, dragFieldID: 1, dropFieldID: null},
            {imgID: 2, img: o_zoovu, dragFieldID: 2, dropFieldID: null},
            {imgID: 3, img: o_zoovu, dragFieldID: 3, dropFieldID: null},
            {imgID: 4, img: v_zoovu, dragFieldID: 4, dropFieldID: null},
            {imgID: 5, img: u_zoovu, dragFieldID: 5, dropFieldID: null}
        ]
        const imagesPositionState = wrapper.state().imagesPosition
        expect(imagesPosition[0]).not.toEqual(imagesPositionState[0])
        expect(imagesPosition[1]).not.toEqual(imagesPositionState[1])
        expect(imagesPosition[2]).not.toEqual(imagesPositionState[2])
        expect(imagesPosition[3]).not.toEqual(imagesPositionState[3])
        expect(imagesPosition[4]).not.toEqual(imagesPositionState[4])
    });

    it('should start timer', function () {
        const wrapper = mount(<Game renderGamePage={true} name={"name"}/>)
        const startTimerMock = wrapper.instance().startTimer = jest.fn();
        wrapper.instance().forceUpdate()
        wrapper.find("img").at(0).simulate('mouseDown', {clientX: 0, clientY: 0}).simulate('mouseMove', {clientX: 0, clientY: 100}).simulate('mouseUp')
        expect(startTimerMock).toHaveBeenCalled()
    });


})


