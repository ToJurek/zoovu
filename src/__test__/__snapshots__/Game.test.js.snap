// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game Componnent render correctly 1`] = `
<div>
  <div
    className="game"
  >
    <div
      className="row"
    >
      <div
        className="column"
      >
        <div
          className="left-side"
        >
          <span>
            Good luck 
            name
            !
          </span>
           
          <br />
          <div
            className="instructions"
          >
            Pick up the right cards
          </div>
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="right-side"
        >
          <span
            className="score"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-clock fa-w-16 clock-icon"
              data-icon="clock"
              data-prefix="far"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200zm61.8-104.4l-84.9-61.7c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h32c6.6 0 12 5.4 12 12v141.7l66.8 48.6c5.4 3.9 6.5 11.4 2.6 16.8L334.6 349c-3.9 5.3-11.4 6.5-16.8 2.6z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            Your score: 
            0
             seconds
          </span>
          <br />
          <div
            className="instructions"
          >
            The faster the better!
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        id="drag2"
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "background": "white",
            "border": "solid #7A7A7A 2px",
            "borderRadius": "28px",
            "flex": "50%",
            "float": "left",
            "height": "150px",
            "margin": "20px",
            "position": "relative",
            "width": "150px",
          }
        }
      >
        <div
          draggable="true"
          id="item2"
          onDragOver={[Function]}
          onDragStart={[Function]}
        >
          <div
            className="sc-AykKC gHzgee"
          >
            <img
              className="img-c"
              id="item2"
              onDragStart={[Function]}
              src={
                Object {
                  "ReactComponent": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                  "__esModule": true,
                  "default": "zoovu-o.svg",
                }
              }
            />
          </div>
        </div>
      </div>
      <div
        id="drag4"
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "background": "white",
            "border": "solid #7A7A7A 2px",
            "borderRadius": "28px",
            "flex": "50%",
            "float": "left",
            "height": "150px",
            "margin": "20px",
            "position": "relative",
            "width": "150px",
          }
        }
      >
        <div
          draggable="true"
          id="item4"
          onDragOver={[Function]}
          onDragStart={[Function]}
        >
          <div
            className="sc-AykKC gHzgee"
          >
            <img
              className="img-c"
              id="item4"
              onDragStart={[Function]}
              src={
                Object {
                  "ReactComponent": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                  "__esModule": true,
                  "default": "zoovu-v.svg",
                }
              }
            />
          </div>
        </div>
      </div>
      <div
        id="drag3"
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "background": "white",
            "border": "solid #7A7A7A 2px",
            "borderRadius": "28px",
            "flex": "50%",
            "float": "left",
            "height": "150px",
            "margin": "20px",
            "position": "relative",
            "width": "150px",
          }
        }
      >
        <div
          draggable="true"
          id="item3"
          onDragOver={[Function]}
          onDragStart={[Function]}
        >
          <div
            className="sc-AykKC gHzgee"
          >
            <img
              className="img-c"
              id="item3"
              onDragStart={[Function]}
              src={
                Object {
                  "ReactComponent": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                  "__esModule": true,
                  "default": "zoovu-o.svg",
                }
              }
            />
          </div>
        </div>
      </div>
      <div
        id="drag1"
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "background": "white",
            "border": "solid #7A7A7A 2px",
            "borderRadius": "28px",
            "flex": "50%",
            "float": "left",
            "height": "150px",
            "margin": "20px",
            "position": "relative",
            "width": "150px",
          }
        }
      >
        <div
          draggable="true"
          id="item1"
          onDragOver={[Function]}
          onDragStart={[Function]}
        >
          <div
            className="sc-AykKC gHzgee"
          >
            <img
              className="img-c"
              id="item1"
              onDragStart={[Function]}
              src={
                Object {
                  "ReactComponent": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                  "__esModule": true,
                  "default": "zoovu-z.svg",
                }
              }
            />
          </div>
        </div>
      </div>
      <div
        id="drag5"
        onDragOver={[Function]}
        onDrop={[Function]}
        style={
          Object {
            "background": "white",
            "border": "solid #7A7A7A 2px",
            "borderRadius": "28px",
            "flex": "50%",
            "float": "left",
            "height": "150px",
            "margin": "20px",
            "position": "relative",
            "width": "150px",
          }
        }
      >
        <div
          draggable="true"
          id="item5"
          onDragOver={[Function]}
          onDragStart={[Function]}
        >
          <div
            className="sc-AykKC gHzgee"
          >
            <img
              className="img-c"
              id="item5"
              onDragStart={[Function]}
              src={
                Object {
                  "ReactComponent": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                  "__esModule": true,
                  "default": "zoovu-u.svg",
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="instruction"
      >
        <span>
          ...and drop them here to make the logo great again!
        </span>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="inputs"
      >
        <div
          id="drop1"
          onDragOver={[Function]}
          onDrop={[Function]}
          style={
            Object {
              "background": "#EAEAEA",
              "border": "green dashed 2px",
              "borderRadius": "28px",
              "flex": "50%",
              "float": "left",
              "height": "150px",
              "margin": "20px",
              "position": "relative",
              "width": "150px",
            }
          }
        />
        <div
          id="drop2"
          onDragOver={[Function]}
          onDrop={[Function]}
          style={
            Object {
              "background": "#EAEAEA",
              "border": "green dashed 2px",
              "borderRadius": "28px",
              "flex": "50%",
              "float": "left",
              "height": "150px",
              "margin": "20px",
              "position": "relative",
              "width": "150px",
            }
          }
        />
        <div
          id="drop3"
          onDragOver={[Function]}
          onDrop={[Function]}
          style={
            Object {
              "background": "#EAEAEA",
              "border": "green dashed 2px",
              "borderRadius": "28px",
              "flex": "50%",
              "float": "left",
              "height": "150px",
              "margin": "20px",
              "position": "relative",
              "width": "150px",
            }
          }
        />
        <div
          id="drop4"
          onDragOver={[Function]}
          onDrop={[Function]}
          style={
            Object {
              "background": "#EAEAEA",
              "border": "green dashed 2px",
              "borderRadius": "28px",
              "flex": "50%",
              "float": "left",
              "height": "150px",
              "margin": "20px",
              "position": "relative",
              "width": "150px",
            }
          }
        />
        <div
          id="drop5"
          onDragOver={[Function]}
          onDrop={[Function]}
          style={
            Object {
              "background": "#EAEAEA",
              "border": "green dashed 2px",
              "borderRadius": "28px",
              "flex": "50%",
              "float": "left",
              "height": "150px",
              "margin": "20px",
              "position": "relative",
              "width": "150px",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
